### PR TITLE
add :wide emote modifier

### PR DESCRIPTION
**Pull request by x-Xymos**
I added a base width of 128px but Dashh said it might be too wide so it's up for debate, anything lower than around 96px stops making a difference to already "wide" emotes like AngelThump

### CODE REVIEW

File: assets/chat/js/const.js [hunk 1]
```diff
@@ -59,7 +59,8 @@ const GENERIFY_OPTIONS = {
     // 'dank': 'generify-dank',
     // 'hyper': 'generify-hyper generify-rustle',
     'love': 'generify-love',
-    'spin': 'generify-spin'
+    'spin': 'generify-spin',
+	'wide': 'wide'
```
Comment (jbpratt78) [inline assets/chat/js/const.js]: Needs tabbing over to match

File: assets/chat/js/formatters.js [hunk 2]
```diff
@@ -192,7 +192,26 @@ class EmoteFormatter {
             
             const innerEmote = ' <span title="' + m + '" class="' + innerClasses.join(' ') + '">' + m + ' </span>';
             const modifierEffect = GENERIFY_OPTIONS[suffix] || "";
-            return ' <span class="generify-container ' + modifierEffect + '">' + hat + innerEmote + '</span>';
+			
+			 let wideEffect = "";
+            if (suffix === "wide"){
+            //scale is relative to the width of the emote
+            const targetWidth = 128; //target width of emote in px
+            if (this.emotewidths[emote] !== undefined) {
```
Comment (jbpratt78) [inline assets/chat/js/formatters.js]: Needs formatting (indention)

**Comment by xDashh** [comment]
Probably have to remove the 3000 line `package-lock.json` file (remove it from the commit and force-push again), except if @KenanY thinks it's a good idea.

Also the indentation is still displayed wrong in the browser-viewer to me.

**Comment by KenanY** [comment]
Yeah please consider undoing the `package-lock.json` changes. Seems like you're using npm v5, which doesn't put semver ranges in the file, where as npm v6 does, so the large diff is v5 removing all the version ranges (inconsequential change but still an unnecessary diff).

File: assets/chat/js/const.js (hunk 1 above)
Comment (KenanY) [reply assets/chat/js/const.js]: Un-resolving this. Should be 4-space indent to match the rest of the file.

**Comment by xDashh** [comment]
So it seems like this modifier was implemented as special case, to my understanding because the css that will be applies varies a bit per emote (depends on emote dimensions). Do you see any easy way to remove that special casing @KenanY ?

**Comment by WhiteNervosa** [comment]
Pardon if this is a dumb interjection because of issues that arise elsewhere, but why not simply put it all in the generified CSS?
```
.wide {
    transform: scaleX(1.5) scaleY(0.8);
    margin: -30px 15px 0px 15px;
    transition: all 0.2s;
}
```

wide

**Comment by x-Xymos** [comment]
> Pardon if this is a dumb interjection because of issues that arise elsewhere, but why not simply put it all in the generified CSS?
> 
> ```
> .wide {
>     transform: scaleX(1.5) scaleY(0.8);
>     margin: -30px 15px 0px 15px;
>     transition: all 0.2s;
> }
> ```
> 
> wide

I can't remember exactly but I was having trouble with some emotes clipping into the text, I tested out your solution and it seems to be working fine except when used with AngelThump it clips a tiny bit. 



I'm not sure if this is ideal or not, if so we can just use the proposed css solution.

**Pull request closed without merging**